auth/auth.go

```diff
@@ -145,3 +177,29 @@ func (a *AuthService) Refresh(w http.ResponseWriter, r *http.Request) {
 		invalidRefreshToken(w)
 	}
 }
+
+func (a *AuthService) oAuthSignUp(data oAuthData, w http.ResponseWriter) {
+	u, findErr := a.client.User.Query().Where(user.EmailEQ(data.GetEmail())).Only(a.ctx)
+	if findErr != nil && !ent.IsNotFound(findErr) {
+		internalServerError(w, fmt.Sprintf("failed to find user: %v", findErr))
+		return
+	}
+
+	if ent.IsNotFound(findErr) {
+		password := a.randomPassword(32)
+		newU, createErr := a.client.User.Create().SetEmail(data.GetEmail()).SetPassword(password).Save(a.ctx)
+		if createErr != nil {
+			internalServerError(w, fmt.Sprintf("failed to create user: %v", createErr))
+			return
+		}
+		u = newU
+	}
```

Comment:
```suggestion
	if ent.IsNotFound(findErr) {
		password := a.randomPassword(32)
		newU, createErr := a.client.User.Create().SetEmail(data.GetEmail()).SetPassword(password).Save(a.ctx)
		if createErr != nil {
			internalServerError(w, fmt.Sprintf("failed to create user: %v", createErr))
			return
		}
		u = newU
	}

	if findErr != nil {
		internalServerError(w, fmt.Sprintf("failed to find user: %v", findErr))
		return
	}
```

By swapping the eror handler clauses it makes one of them just a little bit cleaner.